### DISCUSSION TITLE
Disabled users visible in client gdm

### DISCUSSION THREAD

**Discussion by peter4n** [ltsp/ltsp#203] [Q&A]
I've just installed LTSP19 (from PPA) on Debian 10 and everything is running fine.
However, it looks like that disabled users which are not visible on the server are visible on the client.

The users have been disabled (on the server) with the following command:
`usermod --lock --expiredate 0 <username>`

**Comment by alkisg** [comment]
This is a limitation of the current design (note the old one didn't provide a user list at all).

The information about locked users is stored in the server /etc/shadow. But in that same file, the user passwords are stored in hashed form, and for security reasons we don't want to send them to the clients via unsafe means like TFTP.

So `ltsp initrd` only sends /etc/passwd and /etc/group, *not* /etc/shadow. So the clients do not have the account locking information.

To work around that issue, you need to use the PWMERGE_SUR or PWMERGE_SGR ltsp.conf parameters to exclude those users from appearing on the client. Or you can put such usermod commands in ltsp.conf, with POST_INIT_LOCK_x commands.

Of course, when those users try to login, they will be blocked by SSH as the server will have that information, i.e. it's not a security issue. You could also configure GDM to completely hide the user list.

**Comment by alkisg** [comment]
Well, actually maybe `ltsp initrd` could omit the respective entries from getting sent to the clients, but I feel that would be hackish.

**Comment by peter4n** [comment]
Actually, I was not afraid of an security issue and for me it is no problem at all - I just wanted to mention that issue. Manny thanks to all maintainers for the excellent work.
